internal/model/datacenter.go

```diff
@@ -29,9 +29,9 @@ type Datacenter struct {
 }
 
 // NewDatacenter Initialize datacenter struct
-func NewDatacenter(datacenter *mo.Datacenter) *Datacenter {
+func NewDatacenter(datacenter mo.Datacenter) *Datacenter {
```

Comment:
Why taking the reference to the copy of the object and not of the object of the calling function?

Reply:
No special reason. It's probably just my background in java that makes me think everything is a reference and here it's making another copy. I'll change it

Reply:
And the problem is that when you do a range over a slice you get a "pointer" to the object so in this case I was using that in the function call and the value would be always changes to the last one. So in order to use the "pointer" that range gives you, we needed to pass a copy and then take it's value.

Anyway I changed back to use the index into the slice